### ISSUE TITLE
We need documentation for pre-training BERT with train.py.

### ISSUE THREAD

**Issue by jinho2020** [tensorflow/models#10403]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [v] I checked to make sure that this issue has not been filed already.

## 1. The entire URL of the documentation with the issue

https://github.com/tensorflow/models/blob/master/official/nlp/docs/train.md

## 2. Describe the issue

There is no examples for pre-training BERT.

**Comment by CrickWu** [comment]
Hi @jinho2020! We will try to make these documents more complete. FYI if you want to do a quick test, you can try with the `bert/pretraining` option:

```
python3 train.py \
 --experiment=bert/pretraining \
 --mode=train_and_eval \
 --model_dir=$OUTPUT_DIR \
 --config_file=$CONFIG_FILE \
 --tpu=${TPU_NAME} \
 --params_override=$PARAMS
```

**Comment by saberkun** [comment]
The data preprocessing is a bit painful. You need to use: https://github.com/tensorflow/models/blob/master/official/nlp/data/create_pretraining_data.py at this moment.

**Comment by jinho2020** [comment]
Thanks, @saberkun @CrickWu 
I'm getting a warning on pre-training BERT with train.py.
Is this ignorable or am I missing something on my yaml?

WARNING:tensorflow:Gradients do not exist for variables ['pooler_transform/kernel:0', 'pooler_transform/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss`argument?

**Comment by jinho2020** [comment]
@saberkun @CrickWu 
How can I define the yaml for pre-training BERT?
```yaml
task:
  model:
    encoder:
      type: bert
      bert:
        vocab_size: 32000
```
```yaml
task:
  train_data:
    seq_length: 128
    global_batch_size: 256
    is_training: true
    use_v2_feature_names: false
    use_position_id: false
    use_next_sentence_label: true
    max_predictions_per_seq: 20
  validation_data:
    seq_length: 128
    global_batch_size: 256
    is_training: false
    use_v2_feature_names: false
    use_position_id: false
    use_next_sentence_label: true
    max_predictions_per_seq: 20
trainer:
  optimizer_config:
    learning_rate:
      polynomial:
        decay_steps: 36813
    optimizer:
      type: adamw
    warmup:
      polynomial:
        warmup_steps: 10000
```
